# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_iproute2.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./include/rt_names.h.orig	2003-04-06 17:07:44.000000000 +0200
+++ ./include/rt_names.h	2003-04-06 17:07:52.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef RT_NAMES_H_
 #define RT_NAMES_H_ 1
 
+#include <asm/types.h>
+
 const char* rtnl_rtprot_n2a(int id, char *buf, int len);
 const char* rtnl_rtscope_n2a(int id, char *buf, int len);
 const char* rtnl_rttable_n2a(int id, char *buf, int len);
--- ./include-glibc/netinet/in.h.orig	2003-04-06 17:20:23.000000000 +0200
+++ ./include-glibc/netinet/in.h	2003-04-06 17:23:23.000000000 +0200
@@ -1,3 +1,5 @@
+#include_next <netinet/in.h>
+#if 0
 #ifndef	_NETINET_IN_H
 #define	_NETINET_IN_H	1
 
@@ -12,3 +14,4 @@
 #define SOL_IP	0
 
 #endif	/* netinet/in.h */
+#endif
--- ./include-glibc/netinet/ip.h.orig	2003-04-06 17:20:41.000000000 +0200
+++ ./include-glibc/netinet/ip.h	2003-04-06 17:23:28.000000000 +0200
@@ -1,3 +1,5 @@
+#include_next <netinet/ip.h>
+#if 0
 #ifndef __NETINET_IP_H
 #define __NETINET_IP_H 1
 
@@ -7,3 +9,4 @@
 #include <linux/ip.h>
 
 #endif /* netinet/ip.h */
+#endif
--- ./ip/ipaddress.c.orig	2001-08-23 06:07:46.000000000 +0200
+++ ./ip/ipaddress.c	2003-04-06 17:05:14.000000000 +0200
@@ -20,8 +20,9 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+// #include <linux/netdevice.h>
+#include <net/if.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
@@ -33,6 +34,39 @@
 #include "ll_map.h"
 #include "ip_common.h"
 
+struct net_device_stats
+{
+        unsigned long   rx_packets;             /* total packets received       */
+        unsigned long   tx_packets;             /* total packets transmitted    */
+        unsigned long   rx_bytes;               /* total bytes received         */
+        unsigned long   tx_bytes;               /* total bytes transmitted      */
+        unsigned long   rx_errors;              /* bad packets received         */
+        unsigned long   tx_errors;              /* packet transmit problems     */
+        unsigned long   rx_dropped;             /* no space in linux buffers    */
+        unsigned long   tx_dropped;             /* no space available in linux  */
+        unsigned long   multicast;              /* multicast packets received   */
+        unsigned long   collisions;
+
+        /* detailed rx_errors: */
+        unsigned long   rx_length_errors;
+        unsigned long   rx_over_errors;         /* receiver ring buff overflow  */
+        unsigned long   rx_crc_errors;          /* recved pkt with crc error    */
+        unsigned long   rx_frame_errors;        /* recv'd frame alignment error */
+        unsigned long   rx_fifo_errors;         /* recv'r fifo overrun          */
+        unsigned long   rx_missed_errors;       /* receiver missed packet       */
+
+        /* detailed tx_errors */
+        unsigned long   tx_aborted_errors;
+        unsigned long   tx_carrier_errors;
+        unsigned long   tx_fifo_errors;
+        unsigned long   tx_heartbeat_errors;
+        unsigned long   tx_window_errors;
+
+        /* for cslip etc */
+        unsigned long   rx_compressed;
+        unsigned long   tx_compressed;
+};
+
 static struct
 {
 	int ifindex;
--- ./ip/iptunnel.c.orig	2003-04-06 17:08:43.000000000 +0200
+++ ./ip/iptunnel.c	2003-04-06 17:09:17.000000000 +0200
@@ -16,6 +16,9 @@
  * Phil Karn <karn@ka9q.ampr.org>	990408:	"pmtudisc" flag
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- ./ip/iplink.c.orig	2003-04-06 17:09:55.000000000 +0200
+++ ./ip/iplink.c	2003-04-06 17:10:04.000000000 +0200
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- ./ip/ipmaddr.c.orig	2003-04-06 17:14:25.000000000 +0200
+++ ./ip/ipmaddr.c	2003-04-06 17:14:31.000000000 +0200
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- ./ip/ipmroute.c.orig	2003-04-06 17:14:38.000000000 +0200
+++ ./ip/ipmroute.c	2003-04-06 17:14:43.000000000 +0200
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- ./lib/rt_names.c.orig	2000-04-16 19:42:52.000000000 +0200
+++ ./lib/rt_names.c	2003-04-06 11:36:43.000000000 +0200
@@ -16,6 +16,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <sys/time.h>
+#include <asm/types.h>
 
 static void rtnl_tab_initialize(char *file, char **tab, int size)
 {
--- ./lib/utils.c.orig	2001-08-24 19:09:54.000000000 +0200
+++ ./lib/utils.c	2003-04-06 11:36:43.000000000 +0200
@@ -25,6 +25,7 @@
 #include <netdb.h>
 #include <arpa/inet.h>
 #include <resolv.h>
+#include <asm/types.h>
 #include <linux/pkt_sched.h>
 
 #include "utils.h"
--- ./lib/ll_addr.c.orig	2000-04-16 19:42:52.000000000 +0200
+++ ./lib/ll_addr.c	2003-04-06 11:36:43.000000000 +0200
@@ -17,8 +17,8 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
--- ./lib/ll_proto.c.orig	2001-08-02 04:41:41.000000000 +0200
+++ ./lib/ll_proto.c	2003-04-06 11:36:43.000000000 +0200
@@ -17,8 +17,9 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
+#include <net/if_ether.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
--- ./lib/ll_types.c.orig	2001-08-02 05:18:33.000000000 +0200
+++ ./lib/ll_types.c	2003-04-06 11:36:43.000000000 +0200
@@ -17,8 +17,8 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
--- ./Makefile.orig	2001-08-02 08:37:03.000000000 +0200
+++ ./Makefile	2003-04-06 17:24:39.000000000 +0200
@@ -33,7 +33,7 @@
 
 LDLIBS += -L../lib -lnetlink -lutil
 
-SUBDIRS=lib ip tc
+SUBDIRS=lib ip
 
 LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
 
